File: 3461-check-if-digits-are-equal-in-string-after-operations-i/3461-check-if-digits-are-equal-in-string-after-operations-i.py
class Solution(object):
    def hasSameDigits(self, s):
        """
        :type s: str
        :rtype: bool
        """
        while not self.ispalin(s) and len(s)>2:
            temp=""
            for i in range(1,len(s)):
                temp+=str((int(s[i-1])+int(s[i]))%10)
            s=temp
            
        return self.ispalin(s)

    def ispalin(self,s):
        return s==s[::-1]

File: 3461-check-if-digits-are-equal-in-string-after-operations-i/3461-check-if-digits-are-equal-in-string-after-operations-i.java
class Solution {
    public boolean hasSameDigits(String s) {
        return isPalin(s);
    }

    public boolean isPalin(String s){
        if(s.length()==2) return s.charAt(0)==s.charAt(1);
        return isPalin(adder(s,0));
    }

    public String adder(String str,int index){
        if(index==str.length()-1) return "";
        int a = Character.getNumericValue(str.charAt(index));
        int b = Character.getNumericValue(str.charAt(index + 1));
        int sum = (a + b) % 10;
        return sum+adder(str,index+1);
    }
}